hermes-schema/src/main/java/pl/allegro/tech/hermes/schema/SubjectNamingStrategy.java

```diff
@@ -0,0 +1,18 @@
+package pl.allegro.tech.hermes.schema;
+
+import pl.allegro.tech.hermes.api.TopicName;
+
+public interface SubjectNamingStrategy {
```

Comment:
Very clever and concise self-implementation. :)

Yet, I'm not sure if the exposed API is fine, maybe it would be better to have a simple explicit interface with a single boring implementation and a builder, like:

```java
SubjectNamingStrategy strategy = TopicQualifiedNameSubjectNamingStrategy.builder()
                .withNamespacePrefixIf(() -> true, "kafka")
                .withSuffixIf(() -> true, "-version")
                .build();
// strategy.subjectName(topicName);
```

But I'm not sure, I'll just leave this comment here. :)

Reply:
You know, it is said that "Scala killed the strategy pattern" and Java 8 with its _single abstract method_ followed. :) So since a strategy is no longer an object but a functional interface, the builder pattern hardly applies.

Reply:
I mean this could be implemented as an object. So the final `SubjectNamingStrategy` instance would not have pseudo-builder methods in it's API, just a single `subjectName(topicName)` method. But it's a minor problem, I like it anyways. :)